serializer/serix/map_encode.go

```diff
@@ -145,7 +147,20 @@ func (api *API) mapEncodeStruct(
 	ctx context.Context, value reflect.Value, valueI interface{}, valueType reflect.Type, ts TypeSettings, opts *options,
 ) (any, error) {
 	if valueTime, ok := valueI.(time.Time); ok {
-		return valueTime.Format(time.RFC3339Nano), nil
+		// Convert the int64 timestamp to uint64 by truncating.
+		var unixNano int64
+		// Times whose unix timestamp in seconds would be larger than what fits into a
+		// nanosecond-precision int64 timestamp will be truncated to the max value.
+		if valueTime.Unix() > serializer.MaxNanoTimestampInt64Seconds {
+			unixNano = math.MaxInt64
+		} else {
+			unixNano = valueTime.UnixNano()
+			// Times before the Unix Epoch will be truncated to the Unix Epoch.
+			if unixNano < 0 {
+				unixNano = 0
+			}
+		}
```

Comment:
This is duplicated logic with `WriteTime` in the serializer. Can we instead make a func in the serializer that does the `time.Time -> uint64` conversion and then call that from here and in `WriteTime`?